### DISCUSSION TITLE
[Android] How to handle different email for subscription and the one used to the app?

### DISCUSSION THREAD

**Discussion by therealaezel404** [hyochan/react-native-iap#2826] [Q&A]
**### react-native-iap version: **
12.15.1

**### react-native version:  **
0.74.2

Hi everyone,

I'm working on a react-native project using react-native-iap, and I've encountered a situation where the email used for a subscription might be different from the registered email in my app.

Here's what I'm trying to figure out:

**How should I handle situations where the subscription email and the app's registered email don't match?
What’s the best way to sync the subscription status with the correct user account?
Are there any best practices or recommended approaches for managing this scenario?**

Any guides, examples or advice would be greatly appreciated!

Thanks in advance!

**Comment by hyochan** [comment]
User's Google Play account email may differ from the email registered in your app.

## Solution: Use obfuscatedAccountIdAndroid

Pass your app's user ID when making purchases to link the subscription to your user:

```typescript
import { requestPurchase } from 'react-native-iap';

await requestPurchase({
  request: {
    google: {
      skus: ['subscription_id'],
      obfuscatedAccountIdAndroid: hashUserId(user.id), // Your app's user ID (hashed)
      obfuscatedProfileIdAndroid: hashProfileId(profile.id), // Optional: profile ID
    },
  },
  type: 'subs',
});
```

## How It Works

1. **At purchase time**: Pass your user's ID via `obfuscatedAccountIdAndroid`
2. **Google stores this**: The ID is attached to the purchase record
3. **At validation time**: Retrieve the ID from Google Play Developer API to match with your user

## Server-Side Validation

When validating purchases on your server, the Google Play Developer API response includes:

```json
{
  "obfuscatedExternalAccountId": "hashed_user_id",
  "obfuscatedExternalProfileId": "hashed_profile_id"
}
```

Use this to link the purchase to the correct user in your database.

### Easy Option: Use IAPKit

If you don't want to build your own server, [IAPKit](https://iapkit.com) handles this automatically:

```typescript
import { verifyPurchaseWithProvider } from 'react-native-iap';

const result = await verifyPurchaseWithProvider({
  provider: 'iapkit',
  iapkit: {
    apiKey: 'your-iapkit-api-key',
    google: { purchaseToken: purchase.purchaseToken! },
  },
});
```

## Best Practices

1. **Always hash user IDs** - Don't send plain user IDs to Google
2. **Store mapping on your server** - Map `purchaseToken` → `userId` at purchase time
3. **Use server-side validation** - Don't rely solely on client-side data
4. **Handle account switching** - Users may sign into different Google accounts

## Note on Version

You're using v12.15.1 which is **no longer supported**. Please upgrade to v14+:

- [Migration Guide (v13 to v14)](https://hyochan.github.io/react-native-iap/docs/guides/migration-v13-to-v14)

## References

- [Google: Link purchases to users](https://developer.android.com/google/play/billing/security#link-purchases)
